### PR TITLE
Kindly support

### PR DESCRIPTION
**Pull request by daslu**
(**edit:** added details regarding deps)
(**edit:** fixed temporary HTML link)

This PR It allows to view Wolframite graphics in [Kindly](https://scicloj.github.io/kindly-noted/kindly)-compatible tools.

Changes:
* deps: added Kindly
* extra-deps: added Clay, updated Portal for Clay compatibility
* added a `wolframite.tools.hiccup/view` function - similar to the existing `view*` function but adding a `kind/hiccup` annotation.
* Clay setup: added a `clay.edn` configuration file  and updated `.gitignore`
* added a `dev/kindly_demo.clj` demo notebook, similar to `dev/demo.clj` but using `view`.
* rendered it under a `docs` directory - available temporarily under ~~https://daslu.github.io/wolframite/kindly_demo~~ https://daslu.github.io/wolframite/generated/kindly_demo

Zulip discussion:
https://clojurians.zulipchat.com/#narrow/stream/313853-wolfram-clojure-bridge/topic/Kindly-compatible.20setup

### CODE REVIEW

File: dev/kindly_demo.clj [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+(W:Plus 1 2 3) ; ... and call it
+
+(wl/clj-intern 'Plus {}) ; a simpler way to do the same -> fn Plus in this ns
+
+(run! wl/clj-intern ['Dot 'Plus])
+
+(md " Call interned Wl functions:")
+(Dot [1 2 3] [4 5 6])
+(Plus 1 2 3)
+(Plus [1 2] [3 4])
```
Comment (holyjak) [inline dev/kindly_demo.clj:66]:
```suggestion
```

Rm to keep aligned with #26

File: docs/.clay_files/html-default0.js
Comment (holyjak) [inline docs/.clay_files/html-default0.js:1]: Are the .clay_files / files necessary? (I know 0 about clay)

File: docs/kindly_demo_files/html-default0.js
Comment (holyjak) [inline docs/kindly_demo_files/html-default0.js:1]: I wonder if we should perhaps have these under `docs/generated/kindly_demo*`, to make it clear this are generated from something else? It is also a question whether we want to have them in git at all, and not just generated when we build the site for the project (in the future...). WDYT, @daslu and @light-matters ?

File: docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn
Comment (holyjak) [inline docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn:1]: why do we need this file here?

File: docs/src/wolframite/doc/tutorial.clj.part
Comment (holyjak) [inline docs/src/wolframite/doc/tutorial.clj.part:1]: why is this file here? IS it b/c you renamed doc to docs, and it was always there? (`git mv doc docs` would have made there clear and the git history much nicer)

File: src/wolframite/tools/hiccup.clj [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
+        (img b64img)]]])))
+
+(defn view
+  "View a given Wolframite `form` as Hiccup, Kindly compatible.
+  `:folded true` will fold the view."
+  ([form]
+   (view form nil))
+  ([form {:keys [folded?]}]
+   (-> form
+       (wolframite.tools.hiccup/view* folded?)
```
Comment (holyjak) [inline src/wolframite/tools/hiccup.clj:36]:
```suggestion
       (view* folded?)
```

**Review by holyjak — COMMENTED** [review]
This looks wonderful, thanks a lot!

I have yet to try to run it to see that it works on my machine, but in general LGTM. Though I have a few questions.

BTW, you are a member of the team, you could have just created a branch directly in Wolframite rather than in your fork, which would have made it easier for me to switch to it 😅 #lazy

File: dev/kindly_demo.clj [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+;; This notebook demonstrates basic usage of [Wolframite](https://github.com/scicloj/wolframite/) in a way that would work in visual tools supporting [Kindly](https://scicloj.github.io/kindly-noted/kindly).
+
+;; It is mostly copied and adpated from [the official Wolframite demo](https://github.com/scicloj/wolframite/blob/main/dev/demo.clj).
+
+(ns kindly-demo
+  (:refer-clojure
+   ;; Exclude symbols also used by Wolfram:
+   :exclude [Byte Character Integer Number Short String Thread])
+  (:require
+   [wolframite.core :as wl :refer [wl]]
```
Comment (holyjak) [inline dev/kindly_demo.clj:12]: pls use wl/eval instead of wl/wl (fixed in #26)

File: docs/.clay_files/html-default0.js
Comment (daslu) [reply docs/.clay_files/html-default0.js:1]: `.clay_files`, are not necessary, thank you for catching that.
I now removed them and added to `.gitignore`.

`kindly_demo_files` are necessary.

File: docs/kindly_demo_files/html-default0.js
Comment (daslu) [reply docs/kindly_demo_files/html-default0.js:1]: I think it is a good idea to have them under `docs/generated/`. Maybe we should adopt this convention in other Clay projects.
I'll change this.

Possibly you wish to link to an up-to-date rendered demo from the README. But indeed, it does not have to be under this project. There already is [a draft](https://scicloj.github.io/clojure-data-scrapbook/projects/math/wolframite/index.html) under the [Clojure Data Scrapbook](https://scicloj.github.io/clojure-data-scrapbook/), and maybe that is all we need at the moment.

File: docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn
Comment (daslu) [reply docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn:1]: It is not needed. It is a Clay bug. By default, Clay copies all files under `src` an `notebooks`, except for `clj` files, to the target (`docs` in our case). It is a handy way to allow referring to data, images, etc in a seamless way. Sometimes other things are copied. 

I'll fix it, thanks so much for catching this.

File: docs/src/wolframite/doc/tutorial.clj.part
Comment (daslu) [reply docs/src/wolframite/doc/tutorial.clj.part:1]: It is a bug, just as you commented about [docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn](https://github.com/scicloj/wolframite/pull/28/files/2083b48e1bc9c8435c64a2fd7953f6d4b209d2be#diff-1f3cd3d6e6905074e9989e6492f4a4f341f7e2a5e8dfd1df9ad4e2ec3a087a00). Thanks

File: dev/kindly_demo.clj (hunk 3 above)
Comment (daslu) [reply dev/kindly_demo.clj:12]: Thanks

**Comment by daslu** [comment]
Thanks for all your comments, @holyjak. It is extremely helpful to have this careful look of yours at the Clay setup and find those little bugs.

I'll fix and let you know.

> BTW, you are a member of the team, you could have just created a branch directly in Wolframite rather than in your fork, which would have made it easier for me to switch to it 😅 #lazy

Thanks, this is a habit I have, indeed overcomplicating a bit. And here it was unnecessary, you are right.
(I use a branch in a fork under `daslu` rather than `scicloj` so that I can temporarily set the repo to serve the HTML from a branch different than `main`.
But in this case, indeed I could have done it at `scicloj/wolframite` anyway, as it does not use github pages yet, so does not need to serve from `main` yet.)

File: docs/kindly_demo_files/html-default0.js
Comment (daslu) [reply docs/kindly_demo_files/html-default0.js:1]: Moved to `docs/generated`, thanks.
Also, configured Clay to always clean up this subdirectory on render.

File: docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn
Comment (daslu) [reply docs/src/clj-kondo.exports/scicloj/Wolframite/config.edn:1]: I now configured the call to `clay/make!` (under the `comment` at `kindly-demo` namespace) to avoid syncing those files. Need to think how to avoid it by default.

File: docs/src/wolframite/doc/tutorial.clj.part
Comment (daslu) [reply docs/src/wolframite/doc/tutorial.clj.part:1]: I now configured the call to `clay/make!` (under the `comment` at `kindly-demo` namespace) to avoid syncing those files. Need to think how to avoid it by default.

File: dev/kindly_demo.clj (hunk 3 above)
Comment (daslu) [reply dev/kindly_demo.clj]: done

**Comment by daslu** [comment]
fixed all your comments now, thanks

**Review by holyjak — APPROVED** [review]
Thank you!
